internal/buffer/save.go

```diff
@@ -150,6 +150,32 @@ func (b *Buffer) saveToFile(filename string, withSudo bool, autoSave bool) error
 	// Removes any tilde and replaces with the absolute path to home
 	absFilename, _ := util.ReplaceHome(filename)
 
+	// Resolve possible symlinks to their absolute path
+	for {
+		// Unfortunately only Stat() checks for recursion
+		_, err := os.Stat(absFilename)
+		if err != nil {
+			return err
+		}
+		fileInfo, err := os.Lstat(absFilename)
+		if !os.IsNotExist(err) {
+			if err != nil {
+				return err
+			}
+			if fileInfo.Mode()&os.ModeSymlink != 0 {
+				dstFilename, err := os.Readlink(absFilename)
+				if err != nil {
+					return err
+				}
+				if !filepath.IsAbs(dstFilename) {
+					absFilename = filepath.Join(filepath.Dir(absFilename), dstFilename)
+					continue
+				}
+			}
+		}
+		break
+	}
```

Comment:
> In this regard, a general note about this PR: what about `settings.json` and `bindings.json`? We should write them atomically as well. (So we should not just move this symlink resolving to `overwriteFile()` but also, for example, replace `ioutil.WriteFile()` usage with `overwriteFile()` usage.)

Hm, good point...definitely.
But shouldn't we go the way via a temporary buffer (`NewBufferFromString()`) or will it be too much? From my perspective a lot of additional checks, are done then by the `buffer` layer already.

Reply:
Yeah, and that makes it non-trivial: we need to partially decouple the overwrite functionality from `buffer`, to make it more generic to allow using it not just for buffers but also for other files, like `settings.json`.

And that decoupled functionality would include both the rename (with resolving symlinks etc) and the overwrite fallback.

And since it would include the overwrite fallback, it means we'd want to be able to restore a broken `settings.json` from the saved backup copy, which means that the backup functionality would also need to be partially decoupled from `buffer`?

I'd like to keep it simpler, but how?

Reply:
Currently I can only imagine to do this with a kind of these function literals as callbacks to be performed in the caller.